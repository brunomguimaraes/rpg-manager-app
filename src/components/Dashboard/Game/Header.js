// @flow

import React from "react";
import { Link } from "react-router-dom";
import styles from "./Game.module.scss";
import type { HeaderProps as Props } from "./types";

export default function Header({ gameId, gameName }: Props) {
  return (
    <Link to={`/game/${gameId}`} className={styles.link}>
      <header className={styles.header}>
        <h3>{gameName}</h3>
      </header>
    </Link>
  );
}
