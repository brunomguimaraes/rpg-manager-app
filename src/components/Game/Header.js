// @flow

import React from "react";
import styles from "./Game.module.scss";
import type { HeaderProps as Props } from "./types";

export default function Header({ gameId, gameName }: Props) {
  return (
    <header className={styles.header}>
      <h3>{gameName}</h3>
    </header>
  );
}
