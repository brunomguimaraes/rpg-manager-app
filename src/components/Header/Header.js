// @flow

import React from "react";
import { Link } from "react-router-dom";
import Logo from "~/components/Logo";
import styles from "./Header.module.scss";

export default function Header() {
  return (
    <nav className={styles.header}>
      <ul>
        <li className={styles.logo}>
          <Logo />
        </li>
        <li className={styles.right}>
          <Link className={styles.logo} to="/game">
            New Game
          </Link>
        </li>
      </ul>
    </nav>
  );
}
