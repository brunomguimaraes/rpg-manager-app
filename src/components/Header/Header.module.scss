@import "~styles/variables";

.header {
  font-size: 20px;
  box-sizing: border-box;
  background: $color1;
  ul {
    margin: 0;
    padding: 0;
    display: block;
    overflow: hidden;
    list-style-type: none;
  }
  li {
    float: left;
    display: list-item;

    a {
      color: $color4;
      display: flex;
      align-items: center;
      text-align: center;
      text-decoration: none;
      padding: $spacer * 2 $spacer * 2;
      @media screen and (max-width: $break-small) {
        padding: 12px $spacer * 2;
      }
    }

    &.logo a {
      padding: 11px $spacer * 2 12px;
    }
  }
}

li.right {
  float: right;
}
