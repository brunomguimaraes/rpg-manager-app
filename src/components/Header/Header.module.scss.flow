// @flow
/* This file is automatically generated by css-modules-flow-types */
declare module.exports: {|
  +'header': string;
  +'logo': string;
  +'right': string;
  +'subHeader': string;
  +'subHeaderContainer': string;
|};
