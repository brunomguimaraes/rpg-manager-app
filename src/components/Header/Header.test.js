import React from "react";
import Header from "./Header";

test("Header", () => expect(shallow(<Header />)).toMatchSnapshot());
