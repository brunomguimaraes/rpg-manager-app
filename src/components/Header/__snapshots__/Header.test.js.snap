// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<nav
  className="header"
>
  <ul>
    <li
      className="logo"
    >
      <Logo />
    </li>
    <li
      className="right"
    >
      <Link
        className="logo"
        replace={false}
        to="/game"
      >
        New Game
      </Link>
    </li>
  </ul>
</nav>
`;
